src/Auth0Client.ts

```diff
@@ -708,7 +707,34 @@ export default class Auth0Client {
     const federatedQuery = federated ? `&federated` : '';
     const url = this._url(`/v2/logout?${createQueryParams(logoutOptions)}`);
 
-    window.location.assign(`${url}${federatedQuery}`);
+    return `${url}${federatedQuery}`;
+  }
+
+  /**
+   * ```js
+   * auth0.logout();
+   * ```
+   *
+   * Clears the application session and performs a redirect to `/v2/logout`, using
+   * the parameters provided as arguments, to clear the Auth0 session.
+   * If the `federated` option is specified it also clears the Identity Provider session.
+   * If the `localOnly` option is specified, it only clears the application session.
+   * It is invalid to set both the `federated` and `localOnly` options to `true`,
+   * and an error will be thrown if you do.
+   * [Read more about how Logout works at Auth0](https://auth0.com/docs/logout).
+   *
+   * @param options
+   */
+  public logout(options: LogoutOptions = {}) {
+    const { localOnly } = options;
+
+    if (localOnly) {
+      return;
+    }
```

Comment:
When a developer calls `logout({ localOnly: true })` I would still expect all the logic that clears state and cookies to run. This code as it is here would simply return out of the function without doing anything, and the user would still be logged in.

The point of `localOnly` is that it logs the user out of the app by clearing cookies and internal state, but does not redirect to the `/v2/logout` endpoint at Auth0 (thus keeping the user on the page).

I see all the logic is present in `buildLogoutUrl` - would suggest that logic remains in the `logout` function, and `buildLogoutUrl` simply concerns itself with building a string and producing no side effects.

This probably means that for your scenario in the context of your Capacitor app, you probably want to be doing this in order to log out:

```js
const logoutUrl = client.buildLogoutUrl();
client.logout({ localOnly: true });

// do redirect to `logoutUrl`
```